src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -295,13 +295,19 @@ private static unsafe string Ctor(ReadOnlySpan<char> value)
 
         public static string Create<TState>(int length, TState state, SpanAction<char, TState> action)
         {
-            ArgumentNullException.ThrowIfNull(action);
+            if (action is null)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.action);
+            }
```

Comment:
Why this change, particularly given it changes the exception type?

Reply:
Changing the exception type is a typo; that I need to fix.  This pattern results in ever so slightly smaller asm, and as it's in a generic method and we already had the helper and the appropriately named constant argument, it was trivial to switch.

Reply:
Probably worth a small comment in code so this isn't undone by some future refactoring without validating the change